perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3673,11 +3773,16 @@ private void updateMemberStatus(PerunSession sess, RichMember memberToUpdate) th
 	 * @param mergeMemberAttributesList list of attributes to be merged for member if found
 	 * @param skippedMembers list of not successfully synchronized members
 	 */
-	public void addMissingMemberWhileSynchronization(PerunSession sess, Group group, Candidate candidate, List<String> overwriteUserAttributesList, List<String> mergeMemberAttributesList, List<String> skippedMembers) {
+	public void addMissingMemberWhileSynchronization(PerunSession sess, Group group, CandidateSync candidate, List<String> overwriteUserAttributesList, List<String> mergeMemberAttributesList, List<String> skippedMembers) {
 		Member member;
 		try {
 			// Check if the member is already in the VO (just not in the group)
-			member = getPerunBl().getMembersManagerBl().getMemberByUserExtSources(sess, getPerunBl().getGroupsManagerBl().getVo(sess, group), candidate.getUserExtSources());
+			List<UserExtSource> list = new ArrayList<>();
+			for (RichUserExtSource richUserExtSource : candidate.getRichUserExtSources()) {
+				UserExtSource asUserExtSource = richUserExtSource.asUserExtSource();
+				list.add(asUserExtSource);
+			}
+			member = getPerunBl().getMembersManagerBl().getMemberByUserExtSources(sess, getPerunBl().getGroupsManagerBl().getVo(sess, group), list);
```

Comment:
What if the user can be found only by the user ext source attribute?
Like situation when login in one of the additional user ext-sources is not the same but one of the attributes is?